packages/tableofcontents/init.lua

```diff
@@ -77,7 +82,8 @@ local function _nodesToText (nodes)
       SU.warn("Some content could not be converted to text: "..node)
     end
   end
-  return string
+  -- Trim leading and trailing spaces, and simplify internal spaces.
+  return string:match("^%s*(.-)%s*$"):gsub("%s+", " ")
```

Comment:
```suggestion
  return string:match("^%s*(.-)%s*$"):gsub("%s+", " ")
  return pl.stringx.strip(string, "%s%):gsub("%s%s+", " ")
```

Probably six of one half dozen of the other, but Penlight's `strip()` uses something faster than `:match()` and the more targeted `:gsub()` that only matches multiples not replacing singles is also faster. It also doesn't do quite the same job so if the intention was to replace tabs or other spaces with spaces then go back to what you had. I don't care about this case since it will be run so few times, go with whatever code makes you happy. Just wanted to point out there are a few more ergonomic / efficient things available to you in Penlight for strings, tables, lists, etc.

Reply:
Interesting, I didn't know `pl.stringx.strip()`!

`pl.stringx.strip(string):gsub("%s%s+", " ")`  does the intended thing (IMHO)

There's no need for a second argument to it (and it does work this way, actually, as it expect characters, not a pattern, when specified). I can't say whether it's faster or not - it uses two string.find and a string.sub, FWIW - but it's certainly more readable.